sdk/test/trace/sampler_benchmark.cc

```diff
@@ -0,0 +1,93 @@
+#include "opentelemetry/sdk/trace/samplers/always_on.h"
+#include "opentelemetry/sdk/trace/samplers/always_off.h"
+#include "opentelemetry/sdk/trace/samplers/parent_or_else.h"
+
+#include <benchmark/benchmark.h>
+
+namespace
+{
+using opentelemetry::sdk::trace::AlwaysOffSampler;
+using opentelemetry::sdk::trace::AlwaysOnSampler;
+using opentelemetry::sdk::trace::ParentOrElseSampler;
+// using opentelemetry::sdk::trace::Decision;
+
+void BM_AlwaysOnSamplerConstruction(benchmark::State &state)
+{
+	while (state.KeepRunning())
+	{
+		benchmark::DoNotOptimize(AlwaysOnSampler());
+	}
+}
+BENCHMARK(BM_AlwaysOnSamplerConstruction);
+
+void BM_AlwaysOffSamplerConstruction(benchmark::State &state)
+{
+	while (state.KeepRunning())
+	{
+		benchmark::DoNotOptimize(AlwaysOffSampler());
+	}
+}
+BENCHMARK(BM_AlwaysOffSamplerConstruction);
+
+void BM_ParentOrElseSamplerConstruction(benchmark::State &state)
+{
+	while (state.KeepRunning())
+	{
+		benchmark::DoNotOptimize(ParentOrElseSampler(std::make_shared<AlwaysOffSampler>()));
+	}
+}
+BENCHMARK(BM_ParentOrElseSamplerConstruction);
+
+} // namespace
+BENCHMARK_MAIN();
+
+void BM_AlwaysOnSamplerShouldSample(benchmark::State &state)
+{
+	// Set up parameters
+	AlwaysOnSampler sampler;
+	opentelemetry::trace::TraceId trace_id;
+	opentelemetry::trace::SpanKind span_kind = opentelemetry::trace::SpanKind::kInternal;
+	using M = std::map<std::string, int>;
+	M m1 = {{}};
+	opentelemetry::trace::KeyValueIterableView<M> view{m1};
+
+	while (state.KeepRunning())
+	{
+		benchmark::DoNotOptimize(sampler.ShouldSample(nullptr, trace_id, "", span_kind, view));
+	}
+}
+BENCHMARK(BM_AlwaysOnSamplerShouldSample);
+
+void BM_AlwaysOffSamplerShouldSample(benchmark::State &state)
+{
+	// Set up parameters
+	AlwaysOffSampler sampler;
+	opentelemetry::trace::TraceId trace_id;
+	opentelemetry::trace::SpanKind span_kind = opentelemetry::trace::SpanKind::kInternal;
+	using M = std::map<std::string, int>;
+	M m1 = {{}};
+	opentelemetry::trace::KeyValueIterableView<M> view{m1};
+
+	while (state.KeepRunning())
+	{
+		benchmark::DoNotOptimize(sampler.ShouldSample(nullptr, trace_id, "", span_kind, view));
+	}
+}
+BENCHMARK(BM_AlwaysOffSamplerShouldSample);
+
+void BM_ParentOrElseSamplerShouldSample(benchmark::State &state)
```

Comment:
I think just benchmarking the `ShouldSample` method doesn't give much useful insight, so I wouldn't mind leaving those benchmark tests out for now.

Reply:
The performance tests were not meant to be part of this PR. I had something incorrectly re-based locally. I've went ahead and removed that file.